### PR TITLE
feat: add redemption quotes (part 2)

### PR DESCRIPTION
**Pull request by janndriessen**
* Adds some more values to the quote

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/lib/hooks/use-best-quote/utils/issuance/index.ts [hunk 1]
```diff
@@ -34,30 +48,74 @@ export async function getEnhancedRedemptionQuote(
...
+
+    const callData = encodeFunctionData({
+      abi: DebtIssuanceModuleV2Abi,
+      functionName: 'redeem',
+      args: [
+        Ethereum2xFlexibleLeverageIndex.address! as Address,
+        indexTokenAmount,
+        sender as Address,
+      ],
+    })
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/issuance/index.ts:64]: The quote functions always prepare these tx's, so already encoding the call data here - if the transaction/trade should happen.

File: src/lib/hooks/use-best-quote/utils/issuance/index.ts [hunk 2]
```diff
@@ -34,30 +48,74 @@ export async function getEnhancedRedemptionQuote(
...
+    const gasCostsInUsd = getGasCostsInUsd(gasCosts, request.nativeTokenPrice)
+    transaction.gasLimit = gasEstimate
+    console.log('gasLimit', transaction.gasLimit.toString())
+
+    const inputTokenAmountUsd =
+      parseFloat(formatUnits(indexTokenAmount, inputToken.decimals)) *
+      inputTokenPrice
+    const outputTokenAmountUsd =
+      parseFloat(formatUnits(indexTokenAmount, inputToken.decimals)) *
+      inputTokenPrice
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/issuance/index.ts:90]: Most of the above works similar to the other quote functions. Check `flashmint` or `0x` for reference.

File: src/lib/hooks/use-best-quote/utils/issuance/index.ts [hunk 3]
```diff
@@ -18,11 +21,22 @@ interface RedemptionQuoteRequest {
...
 }
 
+const contract = '0x04b59F9F09750C044D7CfbC177561E409085f0f3'
+
 export async function getEnhancedRedemptionQuote(
   request: RedemptionQuoteRequest,
-  publicClient: PublicClient
+  publicClient: PublicClient,
+  // Using ethers signer for now but will refactor gas estimator to use viem soon
+  signer: providers.JsonRpcSigner
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/issuance/index.ts:30]: Will be a separate PR at some point - as I currently can't estimate the impact of updating gas estimator now.

**Review by 0xonramp — APPROVED** [review]
👍
